src/engine/database/neo4j.rs

```diff
@@ -805,3 +837,57 @@ impl From<Value> for bolt_proto::value::Value {
         }
     }
 }
+
+impl<GlobalCtx, RequestCtx> TryFrom<bolt_proto::value::Value> for Node<GlobalCtx, RequestCtx>
+where
+    GlobalCtx: GlobalContext,
+    RequestCtx: RequestContext,
+{
+    type Error = crate::Error;
+
+    fn try_from(value: bolt_proto::value::Value) -> Result<Self, Error> {
+        match value {
+            bolt_proto::value::Value::Node(n) => {
+                let type_name = &n.labels()[0];
+                let properties: &HashMap<String, bolt_proto::value::Value> = &n.properties();
+                let props_value = Value::try_from(properties.clone())?;
+                let props = match HashMap::<String, Value>::try_from(props_value.clone()) {
```

Comment:
Would it be possible to get rid of the props_value.clone() here.  I know we have to clone it out of the node properties from bolt_proto's Value type, but after that, it would be great to avoid copies where we can.  Would it work to do an if let and use the reference in `format!` the way you did below?